src/base/logger.h

```diff
@@ -82,8 +82,14 @@ class Logger : public QObject
 
     void addMessage(const QString &message, const Log::MsgType &type = Log::NORMAL);
     void addPeer(const QString &ip, bool blocked, const QString &reason = {});
+    Log::Msg getMessage(int id) const;
+    Log::Peer getPeer(int id) const;
     QVector<Log::Msg> getMessages(int lastKnownId = -1) const;
     QVector<Log::Peer> getPeers(int lastKnownId = -1) const;
+    
+    // ids are sequential, following functions return [first known id, last known id]
+    std::pair<int, int> currentMessageRange() const;
+    std::pair<int, int> currentPeersRange() const;
```

Comment:
I think it is a bad idea to use `std::pair` all over the place. Maybe you can use `IndexRange<int>` or `IndexInterval<int>` if the first one doesn't fit.

Reply:
IMO, it seems to be redundant at all.

Reply:
Seems "retrieve by demand" approach still needs some kind of this info.
Although I would reimplement single message retrieving to be index based (real index instead of id). We just need to provide "Logger::messageCount()" method. This way we avoid multiple index-to-id mappings (currently this PR need two mappings to access any message and additional calculations/checkings to get messages count).
So I suggest that "Logger::getMessage(int index)" just means "get Nth message from current set".

Reply:
We still need to maintain id (as timepoint alternative) so async client (e.g. WebAPI user) can access messages subset starting from some timepoint.

Reply:
done?